app/lib/ability.rb

```diff
@@ -76,57 +78,68 @@ def initialize(user)
-    # break out the role and specify permissions more granularly
+    # HUB CLIENT CONTROLLER PERMISSIONS
+    # overly permissive, need to narrow permissions
+    # break out role and specify permissions when making modifications
     client_role_whitelist = [
       :client_success, :admin, :org_lead, :site_coordinator,
       :coalition_lead, :state_file_admin, :team_member
     ].freeze
 
     if user.role?(client_role_whitelist)
-      can :manage, Client, vita_partner: accessible_groups
+      can :read, Client, vita_partner: accessible_groups
+
+      can [:create, :update, :hub_client_management],
+          Client, vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year }
+    end
+
+    if user.role?([:admin, :org_lead, :site_coordinator])
+      can :unlock, Client, vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year }
     end
 
     if user.greeter?
       can [:update, :read, :hub_client_management],
         Client,
         tax_returns: {
-          current_state: [
-            'intake_ready',
-            'intake_greeter_info_requested',
-            'intake_needs_doc_help',
-          ],
+          current_state: %w[intake_ready intake_greeter_info_requested intake_needs_doc_help],
         },
-        vita_partner: accessible_groups
+        vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year }
 
       can [:update, :read, :hub_client_management],
         Client,
         tax_returns: {
-          current_state: [
-            'file_not_filing',
-            'file_hold',
-          ],
+          current_state: %w[file_not_filing file_hold],
           assigned_user: user,
         },
-        vita_partner: accessible_groups
+        vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year }
     end
 
     # Only admins can destroy clients
     cannot :destroy, Client unless user.admin?
-    can :manage, [
+
+    can [:read], [
+      Note,
       Document,
+      TaxReturn
+    ], client: { vita_partner: accessible_groups }
+
+    can [:create, :update, :destroy], [
+      Note,
+      TaxReturn
+    ], client: { vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year } }
+
+    can [:create, :update, :destroy, :archived, :confirm],
+        Document, client: { vita_partner: accessible_groups, intake: { product_year: Rails.configuration.product_year } }
+
+    can :manage, [
       IncomingEmail,
       IncomingTextMessage,
-      Note,
       OutgoingEmail,
       OutgoingTextMessage,
       SystemNote,
-      TaxReturn,
     ], client: { vita_partner: accessible_groups }
 
-    can :manage, TaxReturnSelection, tax_returns: { client: { vita_partner: accessible_groups } }
-    cannot :manage, TaxReturnSelection, tax_returns: { client: { vita_partner: VitaPartner.where.not(id: accessible_groups) }}
```

Comment:
line above already accomplishes the task of restricting access to accessible groups, this line is redundant